base/char.jl

```diff
@@ -112,22 +125,31 @@ end
 #           not to support malformed or overlong encodings.
 
 """
-    ismalformed(c::AbstractChar) -> Bool
+    Base.ismalformed(c::AbstractChar) -> Bool
 
 Return `true` if `c` represents malformed (non-Unicode) data according to the
 encoding used by `c`. Defaults to `false` for non-`Char` types.
 
-See also [`show_invalid`](@ref).
+Any *non*-malformed `c` can be mapped to an integer codepoint
+by [`codepoint(c)`](@ref); this includes codepoints that are
+not valid Unicode characters ([`isvalid(c)`](@ref) is `false`).
+For example, well-formed characters can include invalid Unicode
+codepoints like `'\\U110000'`, unpaired surrogates such as `'\\ud800'`,
+and can also include overlong encodings ([`Base.isoverlong`](@ref)).
+Malformed data, in contrast, cannot be decoded to a codepoint
+(`codepoint` will throw an exception).
```

Comment:
IIUC Unicode does not define a notion of malformedness separate from validity. Is this the only place that the distinction between malformed and wellformed invalid unicode is defined?

Reply:
malformed-ness is a property of the encoding.

However, the UTF-8 encoding (that `String` and `Char` use) is [specified within the Unicode standard](https://www.unicode.org/versions/Unicode16.0.0/core-spec/chapter-2/#G11165).